packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -42,7 +42,7 @@ function contextToRequestLike(ctx: HttpSigContext): RequestLike {
   return {
     url,
     method: ctx.method,
-    headers: ctx.headers,
+    headers: JSON.parse(JSON.stringify(ctx.headers)),
```

Comment:
is there a chance at all that `ctx.headers` is undefined? if so, we'll get an error. 

I think we should either just do `as Record<string, string | string[]>` or write a type guard to check for `undefined` or `null` values in the `ctx.headers` object